proposals/0370-modifiers.rst

```diff
@@ -122,39 +126,136 @@ Proposed Change Specification
     have a known type if declared with a type signature. Alternatively, the
     modifier may have a top-level type signature.
 
-12. A modifier of type ``Multiplicity`` changes the multiplicity of the following arrow,
-    or following pattern-bound variable of a lambda,
-    or preceding record field.
-    Multiple modifiers of type ``Multiplicity`` on the same arrow are not allowed.
-    Any other use of a modifier is an error.
-
-13. ``-XLinearTypes`` implies ``-XModifiers``.
-
-14. Future modifiers will be put *before* the element they modify. Alternatively,
+12. Future modifiers will be put *before* the element they modify. Alternatively,
     a modifier can be put directly before a syntactic closer or separator, such
     as ``;`` or ``where`` or ``)``.
 
-15. Modifiers with an unknown meaning produce a warning, controlled by
-    ``-Wunknown-modifiers``. They are otherwise ignored. (However, in order to
-    know that a modifier is unknown, it still must be parsed, renamed, and type-checked.)
+13. Modifiers of unknown kind produce an error.
+
+14. Modifiers of known kind but with an unknown meaning produce a warning,
+    controlled by ``-Wunknown-modifiers``. They are otherwise ignored. (However,
+    in order to know that a modifier is unknown, it still must be parsed,
+    renamed, and type-checked.)
+
+LinearTypes
+-----------
+With ``-XLinearTypes``:
+
+* A modifier of type ``Multiplicity`` changes the multiplicity of the following
+  arrow, or following pattern-bound variable of a lambda, or following let or
+  where binding, or preceding record field. Multiple modifiers of type
+  ``Multiplicity`` on the same arrow are not allowed. Any other use of a
+  modifier still has no meaning.
+
+* The ``%1`` modifier is handled as a special case. It's renamed (and so
+  typechecked) the same as ``%One``, even if it appears somewhere that linear
+  modifiers aren't expected. If a user does want the modifier ``1 :: Nat``, they
+  can write it as ``%01``. (Requires ``-XDataKinds``.)
+
+* The linear arrow ``a ⊸ b`` has the same meaning as ```a %1 -> b``. Other
+  modifiers are accepted: ``a %Matchable ⊸ b`` has the same meaning as
+  ``a %Matchable %1 -> b``.
+
+With ``-XNoLinearTypes``, the ``%1`` modifier is not special. It refers to the
+type ``1 :: Nat`` and requires ``-XDataKinds``.
+
+``-XLinearTypes`` implies ``-XModifiers``. But the latter can be explicitly
+disabled with ``-XLinearTypes -XNoModifiers``. This introduces backwards
+compatible behavior:
```

Comment:
My *intent* is for this to be backwards compatible, so that users could keep existing programs working by adding `-XNoModifiers`. I'm not super confident I succeeded.

I don't have a strong opinion on whether we want this temporarily, permanently, or not at all. I lean "not permanently", but it's nice that it gives an easier migration route.